presto-docs/src/main/sphinx/connector/raptorx.rst

```diff
@@ -0,0 +1,206 @@
+
+More generically, consider the following figure that contains a predicate and 3 partitions (``A, B, C``) with stats showing min and max. When the partition stats domain does not have any overlap with the predicate domain (e.g. partition ``A``), we could directly prune this partition without sending splits to workers. If the partition stats is completely contained by the predicate domain (e.g. Partition ``C``), then we don’t need this predicate because it would always hold true for this specific partition, and we could strip the predicate when doing plan comparison. For other partitions that have some overlapping with the predicate, we have to scan the partition with the given filter.
+
+.. image:: https://prestodb.io/wp-content/uploads/frc-1024x146.png
+  :align: center
+  :width: 800px
+  :alt: frc image
+
+File Descriptor and Footer Cache
+--------------------------------
+
+A Presto worker caches the file descriptors in memory to avoid long ``openFile`` calls to remote storage. Also, a worker caches common columnar file and stripe footers in memory. The current supported file formats are ORC, DWRF, and Parquet. The reason to cache such information in memory is due to the high hit rate of footers as they are the indexes to the data itself.
+
+Alluxio Data Cache
+------------------
+
+Alluxio data cache has been introduced in `an earlier post <https://prestodb.io/blog/2020/06/16/alluxio-datacaching/>`_. It is the main feature to deprecate the Raptor connector. A Presto worker caches remote storage data in its original form (compressed and possibly encrypted) on local SSD upon read. If, in the future, there is a read request covering the range that can be found on the local SSD, the request will return the result directly from the local SSD. The caching library was built as a joint effort with `Alluxio <https://www.alluxio.io/>`_ and the Presto open source community.
+
+The caching mechanism aligns each read into 1MB chunks, where 1MB is configurable to be adapted to different storage capability. For example, suppose Presto issues a read with 3MB in length starting with offset 0, then Alluxio cache checks if 0 – 1MB, 1 – 2MB, and 2 – 3MB chunks are already on disk and only fetch those that are not cached. The purging policy is based on LRU. It removes chunks from a disk that has not been accessed for the longest time. The Alluxio data cache exposes a standard Hadoop File System interface to the Hive connector, transparently storing requested chunks in a high-performance, highly concurrent, and fault-tolerant storage engine which is designed to serve workloads at Facebook scale.
```

Comment:
with 3MB in length starting with offset 0 --> for the 3 MB starting at offset 0
fetch --> fetches
that has not --> that have not